lib/network/encoding.go

```diff
@@ -120,8 +125,9 @@ func (am *NetworkMessage) SetError(err error) {
 }
 
 var typeRegistry = make(map[uuid.UUID]reflect.Type)
+var registryLock = new(sync.Mutex)
 
```

Comment:
What happens if `UnmarshalRegisteredType`is called as a go-routine several times? And two (or more) go-routines call `registryLock.Lock()`? registry.DeadLock() ...